src/server/job_recov_db.c

```diff
@@ -388,40 +397,10 @@ job_save_db(job *pjob, int updatetype)
 #ifdef __GNUC__
 /* backtrace() functionality provided in glibc since from version 2.1 */
 #if (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 1)
-		/* let's print the backtrace to identify the faulty scenario */
-		int total_frames; /* total number of frames returned by backtrace() */
-		int frame_num;
-		void *bt_buffer[BACKTRACE_BUF_SIZE];
-		char **bt_strings;
-
-		/* backtrace() returns current stack addresses */
-		total_frames = backtrace(bt_buffer, BACKTRACE_BUF_SIZE);
-		if (total_frames != 0) {
-			log_err(-1, "job_save_db", "----- BACKTRACE START -----");
-			sprintf(log_buffer,"backtrace() has returned %d addresses",total_frames);
-			log_joberr(-1, "job_save_db", log_buffer,  pjob->ji_qs.ji_jobid);
-			/* backtrace_symbols() resolve addresses into strings containing "filename(function+address)",
-			 * this array must be free()-ed at the end.
-			 */
-			bt_strings = backtrace_symbols(bt_buffer, total_frames);
-			if (bt_strings == NULL) {
-				sprintf(log_buffer,"no backtrace_symbols() found");
-				log_joberr(-1, "job_save_db", log_buffer,  pjob->ji_qs.ji_jobid);
-			} else {
-				for (frame_num = 0; frame_num < total_frames; frame_num++) {
-					snprintf(log_buffer, LOG_BUF_SIZE-1, "%s", bt_strings[frame_num]);
-					log_err(-1, "job_save_db", log_buffer);
-				}
-			}
-			free(bt_strings);
-			log_err(-1, "job_save_db", "----- BACKTRACE END -----");
-		} else {
-			sprintf(log_buffer,"backtrace() has not returned any address, might be corrupted");
-			log_joberr(-1, "job_save_db", log_buffer,  pjob->ji_qs.ji_jobid);
-		}
-#endif /*-- GLIBC --*/
-#endif /*-- GNUC --*/
-#endif /*-- WIN32 --*/
+		print_backtrace(pjob->ji_qs.ji_jobid);
```

Comment:
Would be tidier if you move the conditional compilation directives inside the body of `print_backtrace()`